mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala

```diff
@@ -79,9 +79,6 @@ abstract class StreamingLinearAlgorithm[
    * @param data DStream containing labeled data
    */
   def trainOn(data: DStream[LabeledPoint]): Unit = {
-    if (model.isEmpty) {
```

Comment:
Why take out this intent? instead of removing `case None` below, since the model must exist